src/java/htsjdk/samtools/SAMLineParser.java

```diff
@@ -170,6 +177,21 @@ private int parseInt(final String s, final String fieldName) {
         }
         return ret;
     }
+    
+    private int parseFlag(final String s, final String fieldName) {
+        final int ret;
+        final SamFlagField samFlagField = this.samFlagField.orElse(SamFlagField.getSamFlagField(s));
+        try {
+            ret = samFlagField.parse(s);
```

Comment:
I will change it to:
```
    private int parseFlag(final String s, final String fieldName) {
        final SamFlagField samFlagField = this.samFlagField.orElse(SamFlagField.getSamFlagField(s));
        try {
            return samFlagField.parse(s);
        } catch (NumberFormatException e) {
            throw reportFatalErrorParsingLine("Non-numeric value in " + fieldName + " column");
        } catch (SAMFormatException e) {
            throw reportFatalErrorParsingLine("Error in " + fieldName + " column: " + e.getMessage(), e);
        }
    }
```
The reason is for me is to clearly identify which method call may causing the exception to be thrown.